cpp/src/interop/from_arrow_host.cu

```diff
@@ -210,15 +210,38 @@ std::unique_ptr<column> dispatch_copy_from_arrow_host::operator()<cudf::string_v
   // offset and char data columns for us.
   ArrowSchemaView view;
   NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, offset_schema.get(), nullptr));
-  auto offsets_column =
-    this->operator()<int32_t>(&view, &offsets_array, data_type(type_id::INT32), true);
+  auto offsets_column = [&]() {
+    if (schema->type == NANOARROW_TYPE_LARGE_STRING) {
+      return this->operator()<int64_t>(&view, &offsets_array, data_type(type_id::INT64), true);
+    } else if (schema->type == NANOARROW_TYPE_STRING) {
+      return this->operator()<int32_t>(&view, &offsets_array, data_type(type_id::INT32), true);
+    } else {
+      CUDF_FAIL("Unsupported string type", cudf::data_type_error);
+    }
+  }();
   NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, char_data_schema.get(), nullptr));
-  auto chars_column = this->operator()<int8_t>(&view, &char_array, data_type(type_id::INT8), true);
 
+  rmm::device_buffer chars = [&] {
+    if (schema->type == NANOARROW_TYPE_LARGE_STRING) {
```

Comment:
Why is this if needed? Seems the `char_array` should be convertable to a vector/buffer regardless of strings type.

Reply:
Hmm yeah you're right. I was trying to preserve the old behavior for the normal strings path, but the extra bits of `int8_t` overload are indeed irrelevant here since we don't store nulls in the chars column anyway so the simple memcpy approach can be used identically for either type.